mPower2/mPower2/PassiveCollectors.swift

```diff
@@ -355,6 +357,30 @@ class PassiveGaitCollector : NSObject, PassiveLocationTriggeredCollector {
         }
     }
     
+    /// The last geofence location (stored in UserDefaults).
+    var previousGeofenceLocation: CLLocation? {
+        get {
+            guard let lastLocDict = UserDefaults.standard.dictionary(forKey: kLastGeofenceLocationKey) else { return nil }
+            let latitude = lastLocDict[kLatitudeKey] as! CLLocationDegrees
+            let longitude = lastLocDict[kLongitudeKey] as! CLLocationDegrees
+            return CLLocation(latitude: latitude, longitude: longitude)
+        }
+        set {
+            guard newValue != nil
+                else {
+                    UserDefaults.standard.removeObject(forKey: kLastLocationKey)
+                    return
+            }
+            let lastLocDict = [
+                kLatitudeKey: newValue!.coordinate.latitude,
+                kLongitudeKey: newValue!.coordinate.longitude
+            ]
+            UserDefaults.standard.set(lastLocDict, forKey: kLastGeofenceLocationKey)
+            UserDefaults.standard.synchronize()
+        }
+    }
```

Comment:
Is there any PII risk of storing this in the user defaults?

Reply:
No more so than using geofencing in the first place, I would think--this same geofence location will be the one currently set or just exited.